### PR TITLE
RGW Zipper - Cleanup and API doc pass

### PR DESCRIPTION
**Pull request by dang**
This is a first pass of documenting the API for Zipper.  It's a very basic pass, documenting each method and some docs on each object.  During this pass, a number of cleanups were identified, and these are also included. 

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: src/rgw/rgw_sal.h [hunk 1]
```diff
@@ -156,94 +205,157 @@ class ObjectProcessor : public DataProcessor {
...
 
+/**
+ * @brief Base singleton representing a Store or stacking layer
+ *
+ * The Store is the base abstraction of the SAL layer.  It represents a base storage
+ * mechanism, or a intermediate stacking layer.  There is a single instance of a given
+ * Store per RGW, and this Store mediates all access to it's backing.
+ *
+ * A store contains, loosely, @a User, @a Bucket, and @a Object entities.  The @a Object
+ * contains data, and it's associated metadat.  The @a Bucket contains Objects, and
```
Comment (cbodley) [inline src/rgw/rgw_sal.h:227]:
```diff
-associated metadat.
+associated metadata.
```

File: src/rgw/rgw_sal.h [hunk 2]
```diff
@@ -102,7 +102,48 @@ class RGWGetUserStats_CB : public RefCountedObject {
...
 
+/**
+ * @file rgw_sal.h
+ * @brief Base abstractions and API for SAL
+ *
+ * The Store Abstraction Layer is an API that separates the top layer of RGW that handles client
+ * protocols (such as S3 or Swift) from the bottom layer of RGW that interacts with a
+ * backing store.  It allows the creation of multiple backing stores that can co-exist
+ * with a single RGW instance, and allows the creation of stacking layers of translators
+ * that can modify operation as they pass down the stack.  Examples of translators might
```
Comment (cbodley) [inline src/rgw/rgw_sal.h:118]:
```diff
-that can modify operation as they 
+that can modify operations as they 
```

**Review by cbodley — APPROVED** [review]
looks great, some minor corrections

**Comment by cbodley** [comment]
it would be nice to have somewhere to host these rendered docs, and 'github pages' can do this pretty easily. i set up https://cbodley.github.io/nexus/ for my https://github.com/cbodley/nexus repo, and it just relies on having a named branch that stores the files output by doxygen. for ceph, we could do this under a new repo like ceph/rgw-sal-docs

**Review by soumyakoduri — APPROVED** [review]

**Comment by dang** [comment]
jenkins test make check

**Pull request merged**